### PR TITLE
Remove SBioBERT dedicated class

### LINKED ISSUE #275: Remove dedicated class for SBioBERT
## Context

Currently, we have a dedicated class for `SBioBERT`:
https://github.com/BlueBrain/Search/blob/c8a884f634c9e667aea2bbba087a048ccb391dd5/src/bluesearch/embedding_models.py#L117-L128

But this is not necessary since we have the class `SentTransformer`, and we could just use SBioBERT as `sbiobert = SentTransformer('gsarti/biobert-nli')` similarly to what we already do for other models:
https://github.com/BlueBrain/Search/blob/c8a884f634c9e667aea2bbba087a048ccb391dd5/src/bluesearch/embedding_models.py#L508-L514

## To do
- [ ] Remove `SBioBERT` class and use `SentTransformer()` instead.
- [ ] Make sure all tests and DVC pipelines work fine.
- [ ] Release new version for `BlueBrain/Search`.

### PR DESCRIPTION
**Pull request by FrancescoCasalegno**
Fixes #275 .

## Description

Previously we were using a dedicated class for SBioBERT, where in particular we defined a method for the mean pooling with attention mask prescribed [here](https://huggingface.co/sentence-transformers/bert-base-nli-mean-tokens).

This PR reduces the complexity of that solution by leveraging [`UKPLab/sentence-transformers`](https://github.com/UKPLab/sentence-transformers/) and the pretrained model [`gsarti/biobert-nli`](https://huggingface.co/gsarti/biobert-nli) through the interface provided by our `bluesearch.embedding_models.SentTransformers`.

## How to test?

The embeddings produced using `SBioBERT` did not change. You can try to run
```python
from bluesearch.embedding_models import get_embedding_model
sbiobert = get_embedding_model("SBioBERT")
sentences = [  # put whatever you want here
     "The neuron is the basic working unit of the brain.", 
     "A system of cells interlinked within one stem."
]  
preprocessed = sbiobert.preprocess_many(sentences)
embedded = sbiobert.embed_many(preprocessed)
```
using the code on `master` and on `remove_sbiobert_class` (e.g. using `pip install -e .` + `git checkout <branch_name>`).

You will see that the arrays `embedded` are the same. **Note**: preprocessed will not be the same because `sentence-transformers` don't need any extra manual pre-processing and therefore its `preprocess()` method is the identity.

## Checklist

- [x] This PR refers to an issue from the [issue tracker](https://github.com/BlueBrain/Search/issues).
  (if it is not the case, please create an issue first).
- [x] Unit tests added.
  (if needed)
- [x] Documentation and `whatsnew.rst` updated.
  (if needed)
- [x] Type annotations added.
  (if a function is added or modified)
- [x] All CI tests pass.

### CODE REVIEW

File: tests/test_server/test_search_server.py [hunk 1]
```diff
@@ -33,15 +33,14 @@ def search_client(
     """Fixture to create a client for mining_server."""
 
     sbiobert_model_inst = Mock()
-    sbiobert_model_class = Mock()
-    sbiobert_model_class.return_value = sbiobert_model_inst
     sbiobert_model_inst.preprocess.return_value = "hello"
```
Comment (jankrepl) [inline tests/test_server/test_search_server.py:36]: Maybe we can change the variable name to something different? Might be confusing to still use `sbiobert_model..`

File: src/bluesearch/embedding_models.py [hunk 2]
```diff
@@ -492,16 +321,17 @@ def get_embedding_model(model_name_or_class, checkpoint_path=None, device=None):
 
     Parameters
     ----------
-    model_name_or_class : str
+    model_name_or_class :
```
Comment (jankrepl) [inline src/bluesearch/embedding_models.py:324]: So I guess you removed the type information from the docstring on purpose since we have annotations, right?
In that case should we maybe remove the `:`? I am not sure what the convention is.

**Review by jankrepl — APPROVED** [review]
Nice job! Always refreshing to see deleted lines > added lines:)

File: tests/test_server/test_search_server.py (hunk 1 above)
Comment (FrancescoCasalegno) [reply tests/test_server/test_search_server.py]: Renamed the variable as `fake_embedding_model` in 7a66438.

File: src/bluesearch/embedding_models.py (hunk 2 above)
Comment (FrancescoCasalegno) [reply src/bluesearch/embedding_models.py]: Fixed in 1f6eec6. By the way, I noticed that it is necessary to specify the return type of the function in the docstring or Sphinx will get confused. 

For instance, if we set the docstring
```python
   """
    Returns
    -------
    sentence_embedding_model : EmbeddingModel
        The sentence embedding model instance.
   """
```
we correctly get 
Screenshot 2021-03-17 at 14 51 15

but if we set the docstring as 
```python
   """
    Returns
    -------
    sentence_embedding_model
        The sentence embedding model instance.
   """
```
we get instead
Screenshot 2021-03-17 at 14 51 53


@jankrepl so any idea on how to treat the return type in docstrings?

Comment (jankrepl) [reply src/bluesearch/embedding_models.py]: My guess is that there are two valid options
1. `variable name : type`
2. `type`

That is why Sphinx did what it did. Maybe there is another syntax that allows for variable names only.

File: src/bluesearch/embedding_models.py [hunk 3]
```diff
@@ -471,7 +296,11 @@ def compute_database_embeddings(connection, model, indices, batch_size=10):
     return final_embeddings, retrieved_indices
 
 
-def get_embedding_model(model_name_or_class, checkpoint_path=None, device=None):
+def get_embedding_model(
+    model_name_or_class: str,
+    checkpoint_path: Optional[Union[pathlib.Path, str, None]] = None,
```
Comment (pafonta) [inline src/bluesearch/embedding_models.py:301]:
```suggestion
    checkpoint_path: Optional[Union[pathlib.Path, str]] = None,
```

File: src/bluesearch/entrypoint/embedding_server.py [hunk 4]
```diff
@@ -43,10 +43,9 @@ def get_embedding_app():
 
     # Load embedding models
     logger.info("Loading embedding models")
+    supported_models = ["SBERT", "SBioBERT", "BioBERT NLI+STS"]
     embedding_models = {
-        "SBERT": SentTransformer("bert-base-nli-mean-tokens"),
-        "BIOBERT NLI+STS": SentTransformer("clagator/biobert_v1.1_pubmed_nli_sts"),
-        "SBioBERT": SBioBERT(),
+        model_name: get_embedding_model(model_name) for model_name in supported_models
```
Comment (pafonta) [inline src/bluesearch/entrypoint/embedding_server.py:48]: Nice! I like that!

File: src/bluesearch/server/search_server.py [hunk 5]
```diff
@@ -110,32 +110,30 @@ def __init__(
...
         ----------
-        model_name : str
+        model_name :
             The name of the model.
 
         Returns
         -------
-        bluesearch.embedding_models.EmbeddingModel
-            The embedding model class.
+        sentence_embedding_model :
```
Comment (pafonta) [inline src/bluesearch/server/search_server.py:123]: Should it be `sentence_embedding_model : EmbeddingModel` like in `embedding_models.py`?

File: src/bluesearch/server/search_server.py [hunk 6]
```diff
@@ -110,32 +110,30 @@ def __init__(
...
-            ),
-            "BioBERT NLI+STS CORD-19 v1": lambda: SentTransformer(
-                biobert_nli_sts_cord19_v1_model_path
+            "SBioBERT": lambda: get_embedding_model("SBioBERT"),
+            "SBERT": lambda: get_embedding_model("SBERT"),
+            "BioBERT NLI+STS": lambda: get_embedding_model("BioBERT NLI+STS"),
+            "BioBERT NLI+STS CORD-19 v1": lambda: get_embedding_model(
+                "SentTransformer", checkpoint_path=biobert_nli_sts_cord19_v1_model_path
             ),
         }
```
Comment (pafonta) [inline src/bluesearch/server/search_server.py:138]: Note as a reminder:

I feel like we are almost to https://github.com/BlueBrain/Search/issues/286 but there is still an extra step to do.

So, https://github.com/BlueBrain/Search/issues/286 is still to do (not the goal of the current PR).

**Review by pafonta — APPROVED** [review]
Thank you for doing this!

I like that we reach more harmonized code.

There are two small changes (typing + docstring inconsistency) to do IMO before merging but otherwise I have tried the `how to test?` and it works:

```
>>> np.allclose(embedded_master, embedded_pullrequest, rtol=0, atol=1e-06)
True
```

File: src/bluesearch/embedding_models.py (hunk 3 above)
Comment (jankrepl) [reply src/bluesearch/embedding_models.py:301]: Hmm, I am not sure if I get this change, could you explain?

Comment (jankrepl) [reply src/bluesearch/embedding_models.py:301]: Ahh, nice find!

File: src/bluesearch/server/search_server.py (hunk 6 above)
Comment (FrancescoCasalegno) [reply src/bluesearch/server/search_server.py:138]: Yes I was thinking about the same.

File: src/bluesearch/server/search_server.py (hunk 5 above)
Comment (FrancescoCasalegno) [reply src/bluesearch/server/search_server.py:123]: Yes, nice catch!

Comment (FrancescoCasalegno) [reply src/bluesearch/server/search_server.py]: Done in 5203164
